### ISSUE TITLE
jottacloud: Get Sky authentication not working

### ISSUE THREAD

**Issue by nattycleopatra** [rclone/rclone#3818]
#### What is the problem you are having with rclone?
Setting up a new Get Sky (white label rebranded Jottacloud) remote isn't working now, and gives HTTP error 401 after attempting to provide the personal login token with the new auth system introduced in #3711.

Setting up new Get Sky remotes with the jottacloud backend used to work before the old authentication method was deprecated.

I've confirmed that the login token authentication method is working with Get Sky accounts when using the official jotta-cli client.

#### What is your rclone version (output from `rclone version`)
rclone v1.50.2-090-g0ecb8bc2-beta

#### Which OS you are using and how many bits (eg Windows 7, 64 bit)
linux/amd64

####  Which cloud storage system are you using? (eg Google Drive)
Jottacloud/Get Sky

#### The command you were trying to run (eg `rclone copy /tmp remote:tmp`)
`rclone config`

#### A log from the command with the `-vv` flag (eg output from `rclone -vv copy /tmp remote:tmp`)
```
$ ./rclone -vv config
2019/12/20 15:24:25 DEBUG : rclone: Version "v1.50.2-090-g0ecb8bc2-beta" starting with parameters ["./rclone" "-vv" "config"]
2019/12/20 15:24:25 DEBUG : Using config file from "/home/user/.config/rclone/rclone.conf"
Current remotes:

Name                 Type
====                 ====
gdrive               drive

e) Edit existing remote
n) New remote
d) Delete remote
r) Rename remote
c) Copy remote
s) Set configuration password
q) Quit config
e/n/d/r/c/s/q> n
name> jotta
Type of storage to configure.
Enter a string value. Press Enter for the default ("").
Choose a number from below, or type in your own value
 1 / 1Fichier
   \ "fichier"
 2 / Alias for an existing remote
   \ "alias"
 3 / Amazon Drive
   \ "amazon cloud drive"
 4 / Amazon S3 Compliant Storage Provider (AWS, Alibaba, Ceph, Digital Ocean, Dreamhost, IBM COS, Minio, etc)
   \ "s3"
 5 / Backblaze B2
   \ "b2"
 6 / Box
   \ "box"
 7 / Cache a remote
   \ "cache"
 8 / Citrix Sharefile
   \ "sharefile"
 9 / Dropbox
   \ "dropbox"
10 / Encrypt/Decrypt a remote
   \ "crypt"
11 / FTP Connection
   \ "ftp"
12 / Google Cloud Storage (this is not Google Drive)
   \ "google cloud storage"
13 / Google Drive
   \ "drive"
14 / Google Photos
   \ "google photos"
15 / Hubic
   \ "hubic"
16 / JottaCloud
   \ "jottacloud"
17 / Koofr
   \ "koofr"
18 / Local Disk
   \ "local"
19 / Mail.ru Cloud
   \ "mailru"
20 / Mega
   \ "mega"
21 / Microsoft Azure Blob Storage
   \ "azureblob"
22 / Microsoft OneDrive
   \ "onedrive"
23 / OpenDrive
   \ "opendrive"
24 / Openstack Swift (Rackspace Cloud Files, Memset Memstore, OVH)
   \ "swift"
25 / Pcloud
   \ "pcloud"
26 / Put.io
   \ "putio"
27 / QingCloud Object Storage
   \ "qingstor"
28 / SSH/SFTP Connection
   \ "sftp"
29 / Transparently chunk/split large files
   \ "chunker"
30 / Union merges the contents of several remotes
   \ "union"
31 / Webdav
   \ "webdav"
32 / Yandex Disk
   \ "yandex"
33 / http Connection
   \ "http"
34 / premiumize.me
   \ "premiumizeme"
Storage> jottacloud
** See help for jottacloud backend at: https://rclone.org/jottacloud/ **

Edit advanced config? (y/n)
y) Yes
n) No (default)
y/n>
Remote config
Config outdated - refreshing
Generate a personal login token here: https://www.jottacloud.com/web/secure
Login Token> [token from https://www.getsky.no/web/secure]
2019/12/20 15:24:47 Failed to get oauth token: HTTP error 401 (401 Unauthorized) returned body: "{\"error\":\"invalid_grant\",\"error_description\":\"Invalid user credentials\"}"
```

**Comment by buengese** [comment]
There are some hardcoded parameters in the current implementation which might be different for getsky. Could you provide me with a token generated by the getsky webinterface? If you generate 2 tokens and give me the older one it should no longer be valid but it would still help me debugging.

**Comment by nattycleopatra** [comment]
Of course! Sent a Get Sky login token in your direction via email, hopefully that's alright.

**Comment by nattycleopatra** [comment]
I guess the "realm" and "well_known_link" properties are what's interesting:
```
{
  "username": "get-$NUMERIC_ID",
  "realm": "get",
  "well_known_link": "https://www.getsky.no/auth/realms/get/.well-known/openid-configuration",
  "auth_token": "$TOKEN"
}
```

**Comment by buengese** [comment]
@nattycleopatra please take a look at #3831

**Comment by larsvinc** [comment]
@nattycleopatra : I have the same desire - using get sky whitelabeled with rclone. Did you get it to work with the whitelabel version?

**Comment by nattycleopatra** [comment]
@larsvinc Yes, authentication was fixed in #3831. I've been using Get Sky with rclone c65bfa1 for the past couple of weeks without any issues.
